policy/modules/services/glusterfs.te

```diff
@@ -58,16 +72,20 @@ manage_files_pattern(glusterd_t, glusterd_runtime_t, glusterd_runtime_t)
 manage_sock_files_pattern(glusterd_t, glusterd_runtime_t, glusterd_runtime_t)
 files_runtime_filetrans(glusterd_t, glusterd_runtime_t, { dir file sock_file })
 
+can_exec(glusterd_t, glusterd_var_lib_t)
```

Comment:
Is this unavoidable? Writing files it can also execute is an easy arbitrary code execution. Glusterd being on the network makes it a prime target.

Reply:
The execute access is needed for hooks that live in `/var/lib/glusterfs/hooks`. To my knowledge these don't get written to so these can just be made a separate type.